server/src/main/java/org/elasticsearch/features/FeatureData.java

```diff
@@ -52,7 +50,7 @@ public static FeatureData createFromSpecifications(List<? extends FeatureSpecifi
                     );
                 }
 
-                if (hfe.getValue().after(CLUSTER_FEATURES_ADDED_VERSION)) {
+                if (hfe.getValue().after(historicalFeaturesLimitVersion)) {
                     throw new IllegalArgumentException(
```

Comment:
I was struggling to figure out what had failed tests when defining a historical feature for 8.14. This doesn't surface anywhere in the test report, can we make sure this gets logged to make it easier to track down the issue?

Reply:
Let me see if I can add a log in tests for this

Reply:
Umm... it already gets logged, and I see it in the console output, in the test report and in the [gradle build scan](https://gradle-enterprise.elastic.co/s/fi4cdjbzvgsxe/tests/task/:qa:smoke-test-http:javaRestTest/details/org.elasticsearch.http.AutoCreateIndexIT/testCannotAutoCreateIndexWhenDisabled?top-execution=1):
```
java.lang.IllegalArgumentException: Historical feature [ml.nlp_supported] declared by [org.elasticsearch.test.rest.RestTestLegacyFeatures@45fcc420] for version [8.14.0] is not a historical version
```
Do you mean adding the limit version explicitly/rephrasing the message?

Reply:
I see. The exception prevents `ESRestTestCase#testFeatureService` static to be initialized, but `client` was, so we hit the assertion. I wonder if there is a way to say "fail all" and stop subsequent tests from running (pointlessly)

Reply:
Let's replace that `assert` with         
```
assumeTrue("Test feature service initialized", testFeatureServiceInitialized());
```

Reply:
The assert could go into the if clause. It doesn't really fail all the consecutive tests and just skips over, but it's easier to understand I think.

Reply:
Adding a message to the assert (to the assert itself) is a good idea; I can do that, taking note.
I don't know if I would use assumeTrue and show the tests as skipped, not failed. Sure, you _should_ have one failed test at the beginning... IDK, I have a feeling this might not be good :)

Reply:
Then I'd suggest to rethrow an exception observed during initialization for every test